apps/skilavottord/ws/src/app/modules/vehicle/vehicle.resolver.ts

```diff
@@ -16,51 +18,68 @@ export class VehicleResolver {
     private vehicleService: VehicleService,
     @Inject(LOGGER_PROVIDER)
     private logger: Logger,
+    @Inject(forwardRef(() => SamgongustofaService))
+    private samgongustofaService: SamgongustofaService,
   ) {}
 
   @Authorize({ roles: [Role.developer, Role.recyclingCompany] })
   @Query(() => [VehicleModel])
   async skilavottordAllVehicles(): Promise<VehicleModel[]> {
-    const res = await this.vehicleService.findAll()
-    this.logger.info('getAllVehicle response:' + JSON.stringify(res, null, 2))
-    return res
+    const vehicles = await this.vehicleService.findAll()
+    this.logger.info(
+      'getAllVehicle response:' + JSON.stringify(vehicles, null, 2),
+    )
+    return vehicles
   }
 
   @Authorize({ roles: [Role.developer, Role.recyclingFund] })
   @Query(() => [VehicleModel])
   async skilavottordAllDeregisteredVehicles(): Promise<VehicleModel[]> {
-    const res = await this.vehicleService.findAllDeregistered()
-    this.logger.info('getAllVehicle response:' + JSON.stringify(res, null, 2))
-    return res
+    const deregisteredVehicles = await this.vehicleService.findAllDeregistered()
+    this.logger.info(
+      'getAllVehicle response:' + JSON.stringify(deregisteredVehicles, null, 2),
+    )
+    return deregisteredVehicles
   }
 
   @Query(() => VehicleModel)
   async skilavottordVehicleById(
     @Args('permno') permno: string,
   ): Promise<VehicleModel> {
-    const res = await this.vehicleService.findByVehicleId(permno)
+    const vehicle = await this.vehicleService.findByVehicleId(permno)
     this.logger.info(
-      'skilavottordVehicleById response:' + JSON.stringify(res, null, 2),
+      'skilavottordVehicleById response:' + JSON.stringify(vehicle, null, 2),
     )
-    return res
+    return vehicle
   }
 
   @Mutation(() => Boolean)
   async createSkilavottordVehicle(
     @CurrentUser() user: User,
     @Args('permno') permno: string,
-    @Args('type') type: string,
-    @Args('color') color: string,
-    @Args('newRegDate') newReg: Date,
-    @Args('vinNumber') vin: string,
   ) {
+    const vehicle = await this.samgongustofaService.getUserVehicle(
+      user.nationalId,
+      permno,
+    )
+    if (!vehicle) {
+      this.logger.error(
+        `User does not have right to call createSkilavottordVehicle action`,
+      )
+      throw new NotFoundException(`User does not have this vehicle`)
+    }
+
     const newVehicle = new VehicleModel()
-    newVehicle.vinNumber = vin
-    newVehicle.newregDate = newReg
-    newVehicle.vehicleColor = color
-    newVehicle.vehicleType = type
+    newVehicle.vinNumber = vehicle.vinNumber
+    newVehicle.newregDate = parse(
```

Comment:
hmmm ... I think we rather want to save this as a Date in the database ... if we want to parse it for readability, then let's do so in the front-end.

Reply:
We are saving it as a Date but vehicle.firstRegDate is string. That's why we have to parse it here

Reply:
aaah I see 👍